src/iop/channelmixerrgb.c

```diff
@@ -3854,11 +3854,16 @@ void reload_defaults(dt_iop_module_t *module)
       g->delta_E_label_text = NULL;
     }
 
+    const int pos = dt_bauhaus_combobox_get_from_value(g->illuminant, DT_ILLUMINANT_CAMERA);
     if(dt_image_is_matrix_correction_supported(img) && !dt_image_is_monochrome(img))
-      // add back if needed
-      dt_bauhaus_combobox_set_from_value(g->illuminant, DT_ILLUMINANT_CAMERA);
+    {
+      if(pos == -1)
+        dt_bauhaus_combobox_add_introspection(g->illuminant, NULL,
+                                              module->so->get_f("illuminant")->Enum.values,
```

Comment:
I liked the brevity of the previous
```
dt_bauhaus_combobox_set_from_value(g->illuminant, DT_ILLUMINANT_CAMERA);
```
- `dt_bauhaus_combobox_add_introspection` on the other hand seems a bit verbose because one has to dig up the enum tuple and pass it as well (plus the `NULL` action parameter). As a module author I wouldn't be able to very quickly pull the `module->so->get_f("illuminant")->Enum.values` unless looking at other modules. (It feels slightly like this is an "internal" structure that's not supposed to be used directly)

It would be convenient if there was be a variant of `dt_bauhaus_combobox_add_introspection` that looked up the enum tuple list itself, like `dt_bauhaus_combobox_set_from_value` does.

Obviously this is now just aesthetics and splitting hairs, and you may as well ignore this one.

Reply:
`dt_bauhaus_combobox_set_from_value` way of finding the enum is quite roundabout (first looking up the action in a hashtable and then looking up an associated enum list in another hashtable), whereas here we already "know" where to find the table. Agreed with the "internal structure" comment, but also don't necessarily want to normalise changing lists on the fly all the time either? And we already have ` module->get_f("illuminant")->Enum.Default;` and `Float.Default` in places, so if we start introducing helper functions for all of them...

If we _do_ want a helper function for this, it should just be to look up the `dt_introspection_type_enum_tuple_t` because I'd still want the flexibility of `dt_bauhaus_combobox_add_introspection` to be able to add a _series_ of values in one go and that function isn't _just_ used for introspection generated lists (even if they use that format).

Reply:
Fair enough, I don't think there's a need for further helper functions.

Reply:
If this pattern is used more in future (tbh people will learn how to do this by looking at other modules and copy/pasting anyway, no matter how obvious a function is) the need for a helper function/refactoring can always be reviewed. And at that point with more data on what common use patterns are.

Thanks again for reviewing and commenting. Much appreciated!